File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>pascal(numRows);
        pascal[0]={1};
        for(int i=1;i<numRows;i++)
        {
            vector<int>row={1};
            for(int j=1;j<i;j++)
            {
                row.push_back(pascal[i-1][j-1]+pascal[i-1][j]);
            }
            row.push_back(1);
            pascal[i]=row;
        }
        return pascal;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascal = new ArrayList<>();
        for(int i=1;i<=numRows;i++)
        {
            List<Integer> arr = new ArrayList<> (Collections.nCopies(i, 1));
            for(int j=1;j<i-1;j++)
            {
                arr.set(j,pascal.get(i-2).get(j)+pascal.get(i-2).get(j-1));
            }
            pascal.add(arr);
        }
        return pascal;
    }
}